backend/src/report_director.py

```diff
@@ -0,0 +1,21 @@
+
+from typing import TypedDict
+from fastapi import UploadFile
+
+from src.utils.scratchpad import update_scratchpad
+from src.utils.file_utils import handle_file_upload
+
+class FileUploadReport(TypedDict):
+    id: str
+    filename: str | None
+    report: str | None
+
+async def report_on_file_upload(upload:UploadFile) -> FileUploadReport:
+
+    file = handle_file_upload(upload)
+
+    update_scratchpad(result=file["content"])
+
+    report = "#Report on upload as markdown" # await report_agent.invoke(file["content"])
```

Comment:
I am wondering if the director should be responsible for storing the report in the scratchpad, or if the agent should be doing it.

Reply:
Probably the director, though I don't think we have any reason to be using the scratch pad, so it's tempting to just remove completely for now.